### ISSUE TITLE
PowerShell LTS:Continue releasing version for ubuntu1404 please, after one year more than EOL

### ISSUE THREAD

**Issue by chuanjiao10** [PowerShell/PowerShell#9496]
# Continue releasing powershell version to ubuntu1404, after one year or more than Linux distribution eol

Welcome discussion.
Most software does not stop releasing, at the Linux distribution eol day.
Especially powershell is developing.

on the powershell 6.2 release day ,i post news at china group,a debian8 user tell me he can not install 6.2.

Recently, ubuntu 1404 went to eol,so i  hope powershell-team can agree ，give powershell release to that distribution . one years or more.

There are a lot of ubuntu 1404 and debian 8 users. ubuntu 1404 users more than debian 8 users. Would rather not release the version of fedora, but also ubuntu 1404 and debian8 for release.

**Comment by iSazonov** [comment]
A problem is that we need .Net Core support before we can support PowerShell Core.
Please move request to .Net Core repo.

**Comment by chuanjiao10** [comment]
hi @iSazonov .this issue talk about powershell version LTS topic. Not related to .Net Core.Please correct the label.
Why did you close the issue too early?

**At https://github.com/dotnet/core-setup ,see "Debian 8.2 (x64)",** 
A problem is that  .Net Core support debian 8 ,but powershell-team give not PowerShell release now.

ubuntu 1404 users more than debian 8 users.If powershel-lteam doesn't want to release the version to debian8, I don't want to stick to it. But at least it should be released to ubuntu 1404.

**Comment by iSazonov** [comment]
Ubuntu 14.04 is in supported list .Net Core 2.1 https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1-supported-os.md
.Net Core 2.1 is LTS https://dotnet.microsoft.com/platform/support/policy/dotnet-core with end of support August 21, 2021
Ubuntu 14.04 End of Standard Support is April 25, 2019 then only paid support.

This information is in PowerShell docs https://docs.microsoft.com/en-us/powershell/scripting/powershell-support-lifecycle?view=powershell-6#powershell-release-end-of-life

PowerShell Core 6.1 end of life is September 28, 2019. For 6.2 I guess - + 6 months (April, 2020)

So we will get servicing updates for the versions for all _supported_ platforms.

It is nor clear will .Net Core release servicing updates for Ubuntu 14.04 or no.
/cc @SteveL-MSFT @joeyaiello

**Comment by chuanjiao10** [comment]
For powershell lts, my point of view is:
Powershell should not be exactly the same as the .net support cycle.
For these versions, at least one year is postponed and two years are recommended.
Ubuntu 1404 = 1 year. After eol one year.
Ubuntu 1604 = 2 years.
Ubuntu 1804 = 2 years.

Debian8=1 years.
Debian9=2 years.

Centos7=2 years.
Last suse enterprize=2 years.

So my main point of view is:
Let powershell 6.2 give debian8, ubuntu1404, one year powershell software release.----For the going to eol version of Linux distribution , Force users to use the latest version of powershell.

Let powershell 6.1 stop publishing the version for debian8, ubuntu1404.

**Comment by chuanjiao10** [comment]
![6K5$C@H_{ Y9N6YC}5N YO](https://user-images.githubusercontent.com/8968667/56960884-d1218200-6b84-11e9-8707-06054104197c.png)

![XUOFODUC${V9J_2{516)}72](https://user-images.githubusercontent.com/8968667/56960919-e7c7d900-6b84-11e9-93aa-8d369de98559.png)

![X0Y`NZ3H4 N(H5Z_57ZJ4IE](https://user-images.githubusercontent.com/8968667/56960973-0a59f200-6b85-11e9-907c-f636aa7d90ba.png)

**Comment by chuanjiao10** [comment]
Because there is no support rule for eol linux distribution, the powershell version is released at will. 

There 2 easy way:（Here I use debian8 as an example.Treat ubuntu1404 equally.）

1 continue release newest version (6.2) to debian8 (eol linux distribution), until 1 year is full.

2 if  first  powershell ga version （6.1.0 ，or 6.2.0） release for debian8 (eol linux distribution), then continue release powershell version for debian8, until this powershell version eol day.

**Comment by SteveL-MSFT** [comment]
PSCore6 has a different support lifecycle than .NET Core.  However, for PS7, we plan to align with .NET Core LTS-style support.  Since

**Comment by TravisEz13** [comment]
@SteveL-MSFT We should consider the compliance of any such builds in the committee review.

**Comment by TravisEz13** [comment]
DotNet Support for Ubuntu 14.04 ends when the OS EOL's per https://github.com/dotnet/core/blob/master/os-lifecycle-policy.md#net-core-supported-os-lifecycle-policy

**Comment by TravisEz13** [comment]
Another problem, is that we cannot test the releases post EOL, without changes to our infrastructure as Ubuntu takes down their repos.

**Comment by iSazonov** [comment]
We could make servicing branches public (and unsupported by MSFT) so consumers could fork and compile locally if they really need.

**Comment by TravisEz13** [comment]
Servicing releases are made as a tag, people can branch a tag.  There is no reason to keep a branch alive.  This is the same process we follow.

**Comment by chuanjiao10** [comment]
I think,
The powershell committee should first determine a principle.
Which Linux distributions will lose the powershell release (such as fedora) on the day of its eol; 
which delays are 1 year (such as ubuntu1404); 
which delays are 2 years.

Then, make a table listing the above in html.

**Comment by TravisEz13** [comment]
Or keep the current policy where the EoL of the underlying distro determines our EoL.

**Comment by chuanjiao10** [comment]
For release, there is also a suggestion for the name of the linux package: (not to use it does not matter)

Suppose we release 7.0 preview1 , 7.0 preview2 , 7.0 rc1 , 7.0 rc2 ,
Then it is recommended to use the package name powershell-preview-7.0.1.rpm,powershell-preview-7.0.2.rpm, powershell-preview-7.0.3, powershell-preview-7.0.4 ...
 (rc1 package name uses preview3, rc2 package name uses preview4, the order is not easy to make mistakes, because yum does not know "powershell-rc1" is newer than "powershell-prewiew2" .yum don't know if they are a piece of software ) 
The same thing we hinted at the release note, (rc1 packagename is preview3).

**Comment by joeyaiello** [comment]
@chuanjiao10 that list issue definitely sounds like a real one. I think there's other solutions to it, but please file an issue for that one separately. 

As for this issue, I don't understand how we can support something without underlying support in .NET Core and the OS. Sometimes things will "just work", and we won't ever restrict you from using that, but "supported" implies we're keeping pace with security updates, and if there are no underlying security updates, we're leaving our customers vulnerable.

@SteveL-MSFT might have context here that I don't have, though. We'll discuss it today.

**Comment by SteveL-MSFT** [comment]
@PowerShell/powershell-committee reviewed this and agrees that we cannot support unsupported OS' nor unsupported .NET Core releases.  Anyone who wants to use it can build their own and support it themselves.

**Comment by chuanjiao10** [comment]
no no no , my point never at "support"! ，my point never at "keeping pace with security updates"!  "support" is iSazonov say. my point is  "release powershell version after eol day".
Now that the powershell committee has reached a conclusion, I just want to hear the results now. 

**Powershell-team will not release the powershell version for  Linux distribution after them EOL day** , right?

**Comment by chuanjiao10** [comment]
Whether to release the version, regardless of "support" and "keeping pace with security updates"
eg：
on https://www.mongodb.com/download-center/community ，we can download newest version of mongodb software for debian8.

on http://nginx.org/packages/debian/pool/nginx/n/nginx/ , we can download newest version of nginx, the binary package is http://nginx.org/packages/debian/pool/nginx/n/nginx/nginx_1.16.0-1~jessie_amd64.deb                   23-Apr-2019 15:13              835644

**Comment by iSazonov** [comment]
@chuanjiao10 MSFT team haven't resources to release PowerShell for platforms after EOL. 
This is also prohibited by Microsoft security policies.
Some Linux communities release PowerShell distributives by themselves. Ubuntu community can do the same too.

**Comment by chuanjiao10** [comment]
@iSazonov  i known now.
Obviously, after 2020-01-15, ms will release new version to .net framework on win7,
Ms will release new version to .net core on win7,
With the increase in powershell users, powershell will have to do this in the future.

Should not be lazy to give up the existing powershell users!
**Powershell-team will give up release ps version 6&7  to win7 after 2020-01-15?** Time is not far away, let us wtaching.

**Comment by iSazonov** [comment]
> Powershell-team will give up release ps version 6&7 to win7 after 2020-01-15?

If only an interested community I think.
